source/Irrlicht/CIrrDeviceLinux.cpp

```diff
@@ -1178,6 +1248,59 @@ void CIrrDeviceLinux::setWindowCaption(const wchar_t* text)
 }
 
 
+//! Sets the window icon.
+bool CIrrDeviceLinux::setWindowIcon(const std::string &path)

```

Comment:
should take an `IImage` IMO

Reply:
<!--
I'm not sure how to properly do the win32 implementation though.
It currently does some hacky stuff where it ignores the path and uses the `winresource.rc` file (which uses `minetest-icon.ico`). If this function required an IImage, we'd have to be able to open it (in `renderingengine.cpp`), just so that win32 can ignore it. The paths that `renderingengine.cpp` currently uses do probably not even exist in win32.
-->

Done.

Note though that win32 will now also require the (unused by win32) icon image file to work.

Reply:
As of now that doesn't work, the file is not installed on Windows.
(assuming you didn't change that in the relevant PR)

Reply:
(Windows installations would also need the file for the SDL driver.)

>assuming you didn't change that in the relevant PR

I didn't. Idk how to fix this, and I couldn't test.